blog/_src/posts/2016-10-17-emacs-daemon-for-fast-editor-startup.md

```diff
@@ -0,0 +1,63 @@
+    Title: Emacs daemon for fast editor startup
+    Date: 2016-10-17T21:48:25
+    Tags: System Administration, Emacs, by Gabriel Scherer
+
+In the famous Emacs/Vim debates, Emacs was often ridiculed for its
+bulkiness (Eight Megabytes-of-RAM And Constantly Swapping, etc.). The
+computational power of our computer has grown much faster than Emacs'
+bloat: it takes exactly one second to load on my machine. However, our
+workflows have also changed, and my workflow implies frequently
+starting new text editors -- on each git commit for example, or
+when I use a [Firefox
+extension](https://addons.mozilla.org/en-US/firefox/addon/its-all-text/)
+to edit a textarea content in a proper editor.
+
+In this blog post, I describe how to use `emacsclient` to reuse an
+existing Emacs process when creating a new editor window, which
+reduces editor startup times from 1s to 0.150s on my machine.
+
+<!-- more -->
+
+Emacs has long supported a client/server mode: a daemon emacs instance
+is loaded in the background, and whenever you request a new emacs
+window you can creates a new frame (~ window) using the same
+instance. This means that the startup time is dramatically reduced
+after the daemon is launched, as for example the execution of your
+personal configuration code does not have to be repeated.
+
+To use it, I have this code as `/usr/bin/editor`:
+
+```sh
+#!/bin/bash
+emacsclient -a "" -c "$@"
+```
+
+The empty `-a` parameter means that if no daemon exists, it should start one in the background and retry.
+The `-c` option means that a new frame (window) should be created
+instead of reusing an existing one. `"$@"`means that when the script
+is invoked with a path as command-line parameter (`editor /tmp/foo.txt`),
+the corresponding file will be opened.
+
```

Comment:
Do you also `export EDITOR="editor"` in a dotfile?
(Either way, I think it'd be good to say how to get `git commit` to use "fast emacs")